### PR TITLE
feat: `dump` subcommand

### PR DESCRIPTION
**Pull request by suft**
## Summary

Adds a `daylog dump` command that concatenates every log in a project, addressing #82.

- I went with a standalone `dump` subcommand instead
- Since it's project-wide rather than a variant of showing a single day's log
- It leaves room for its own flags (`--reverse`, `--since`, `--until`) without overloading `show`
- Happy to rename/restructure if a flag on `show` is preferred instead.

## What's included

- **`daylog dump`** — concatenates all logs, oldest → newest by default.
- **`--reverse`** — newest → oldest instead.
- **`--since` / `--until`** — restrict the range to logs on or after / on or before a given date (inclusive on both ends)
- **`-o/--output`** — reuses the existing `markdown`/`text` output formatter, same as `show`.
- Respects `-p/--project` like the other commands.

## Example usage

```
daylog dump
daylog dump -r
daylog dump --reverse
daylog dump -o text
daylog dump --since "1 week ago"
daylog dump --since 2023/01/07 --until yesterday
daylog dump -p work
```
Closes #82

### CODE REVIEW

File: internal/daylog/daylog.go [hunk 1]
```diff
@@ -309,6 +310,55 @@ func Search(projectPath, query string, ignoreCase bool) ([]SearchMatch, error) {
...
+		content, err := os.ReadFile(filepath.Join(projectPath, log, "log.md"))
+		if err != nil {
+			// a log removed between listing and reading shouldn't sink the
+			// whole dump; skip it and return what we can
+			if os.IsNotExist(err) {
+				continue
+			}
+			return "", err
+		}
+		b.Write(content)
```
Comment (notnmeyer) [inline internal/daylog/daylog.go]:
```suggestion
		b.WriteString(strings.TrimRight(string(content), "\n"))
		b.WriteString("\n\n")
```

i think theres a mild edge case here where if someone's editor does not append a trailing newline to the file you can wind up with the logs running together like this,

```
➜ daylog dump -p newlines-test

   2025/12/01

  • ate a burrito.# 2025/12/02
  • ate a taco.
```

with this suggestion,
```
➜ daylog dump -p newlines-test

   2025/12/01

  • ate a burrito.

   2025/12/02

  • ate a taco.
```
